### PR TITLE
Plugin: Introduce `tools` folder with configuration files

### PR DESCRIPTION
**Pull request by gziolo**
## Description

This PR proposes a new top-level folder with configuration files for tools:

```
     └── tools/eslint
     │   Configuration files for the ESLint linter.
     │
     └── tools/webpack
     │   Configuration files for the webpack build.
```

It allows two things:
- move linting helper for ESLint from `test/linter` folder to `tools/eslint`
- move all the webpack configuration from the top-level folder to `tools/webpack/packages`

The next step would be to split webpack config into two subconfigs:
- packages
- blocks

It's similar to what's proposed in WordPress core in https://github.com/WordPress/wordpress-develop/pull/1412.

### CODE REVIEW

File: tools/webpack/packages.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+			// as ESM if they use `import` or `export` within it.
+			const request = rootModule?.rawRequest || rawRequest;
+
+			if ( request.includes( '/view.js' ) ) {
+				return `./build/block-library/[name]/view.min.js`;
+			}
+
+			return `./build/[name]/index.min.js`;
+		},
+		path: join( __dirname, '..', '..' ),
```
Comment (gziolo) [inline tools/webpack/packages.js:160]: This is the only code change in comparison to the original `webpack.config.js` file.

**Review by youknowriad — COMMENTED** [review]
This is looking good, I do wonder whether "tools" is the right name though, it feels this is more "config" than "tools", though both are problematic in some ways:

 - tools: what about documentation tool, release tool, api doc tool...
 - config: what about the other configs we have spread everywhere :) 
 -

**Comment by gziolo** [comment]
I followed the pattern that is used in https://github.com/WordPress/wordpress-develop/tree/master/tools

Screen Shot 2021-07-08 at 14 11 14

**Review by youknowriad — APPROVED** [review]
Approving, I'm just nitpicking above. I like that it aligns with core and potentially simplify porting updates to the configs.

**Pull request merged**